code/modules/jobs/job_types/_job.dm

```diff
@@ -133,6 +155,112 @@
+/// in the medical department, and the max limit of geneticist slots available, if the lowpop limit has not been met.
+/// = Params =
+/// ignore_self_limit: bool => If set, then the max limit of the job will be ignored when using the dynamic group
+/// scaling calculations.
+/datum/job/proc/get_spawn_position_count(ignore_self_limit = FALSE)
+	var/player_count = SSjob.initial_players_to_assign
+	// SSjob has not been allocated yet
+	if (!player_count)
+		player_count = length(GLOB.clients)
+	// Out of range, and no proxy
+	if (player_count < min_pop && !min_pop_redirect)
+		return 0
+	if (player_count > max_pop)
+		return 0
+	// Unlimited, though we are limited if we use a dynamic spawn group
+	// We must be:
+	// - Unlimited ourselves
+	// - Not present in a dynamic spawn group
+	// - Not using a proxy due to lowpop, or the proxy has no dynamic spawn group
+	if (total_positions == -1 && !dynamic_spawn_group && (player_count >= min_pop || !min_pop_redirect || !min_pop_redirect::dynamic_spawn_group))
+		return -1
+	// If the population is lower than our min pop spawn amount
+	// then we will instead treat ourselves
+	var/datum/job/proxy = src
+	if (min_pop_redirect && player_count < min_pop)
+		proxy = SSjob.GetJob(min_pop_redirect::title)
+	// Does not have a spawn group
+	if (!proxy.dynamic_spawn_group)
+		// The proxy role allows for infinite joining, so we do too
+		if (proxy.total_positions == -1)
+			return -1
+		return max(proxy.total_positions + total_position_delta, 0)
+	// Calculate spawn group size
+	var/spawn_group_minimum = INFINITY
+	for (var/datum/job/other in SSjob.occupations)
+		// Find everything in the same group, doesn't matter if its us
+		if (other.dynamic_spawn_group != proxy.dynamic_spawn_group)
+			continue
+		// Find the least filled job in the group
+		// If the HOP removes a position from another job, then that removed position.
+		// If the HOP adds a position to a job group, then it has to be filled before the spawn
+		// group bumps.
+		spawn_group_minimum = min(spawn_group_minimum, other.get_group_position_count())
+	// The amount of positions we have is the least filled job + our allowed variance
+	// variance is calculated per job, not based on the proxy
+	// If we are using a proxy, then the number of spawn positions is limited to the total
+	// positions available in that role, regardless of whether or not the department as a whole
+	// has extra space.
+	// If the proxying target has a dynamic spawn group, then that implictly means that there is
+	// no limit to the total positions; this behaviour only exists so that we can limit the number
+	// of players joining as a sub-role of a department, not as the primary role.
+	return min((proxy == src || ignore_self_limit || proxy.dynamic_spawn_group) ? INFINITY : total_positions, max(spawn_group_minimum + proxy.dynamic_spawn_variance_limit + total_position_delta, 0))
```

Comment:
```suggestion
	var/position_count_a = (proxy == src || ignore_self_limit || proxy.dynamic_spawn_group) ? INFINITY : total_positions
	var/position_count_b = max(spawn_group_minimum + proxy.dynamic_spawn_variance_limit + total_position_delta, 0)
	return min(position_count_a, position_count_b)
```
I believe merging all conditions into a single line has less ideal for reading, and merging them into one line has very minimal gain for the performance.
btw does `INFINITY` has any point?